addons/stock/models/stock_location.py

```diff
@@ -77,7 +77,7 @@ def default_get(self, fields):
     last_inventory_date = fields.Date("Last Effective Inventory", readonly=True, help="Date of the last inventory at this location.")
     next_inventory_date = fields.Date("Next Expected Inventory", compute="_compute_next_inventory_date", store=True, help="Date for next planned inventory based on cyclic schedule.")
     warehouse_view_ids = fields.One2many('stock.warehouse', 'view_location_id', readonly=True)
-    warehouse_id = fields.Many2one('stock.warehouse', compute='_compute_warehouse_id', recursive=True)
+    warehouse_id = fields.Many2one('stock.warehouse', compute='_compute_warehouse_id')
```

Comment:
What about storing it  in master ?

Reply:
Mhhh, it allows to search/groupby/read it more efficiently. Why not ? The only boring stuff is that we should modifies my invalidated cache to avoid recompute every location warehouse_id when we create or modify 1 location. 
New task maybe ?

Reply:
> New task maybe ?

Fais toi plaiz'